### ISSUE TITLE
How to reject or cancel the promise after some time(say 6 after seconds). How to timeout odoo call.

### ISSUE THREAD

**Issue by abhijeetgiram** [karthikrp/angular6-odoo-jsonrpc#3]
this.odooRPC.call('model_name, 'method_name', parameter, {})
      .then(res => {
        console.log('call success', res);
      })
      .catch(err => {
        console.error('call failed', err);
      });

I have called "method_name" in the above way. How can I timeout the same like HTTP timeout? Need help. Urgent.

**Comment by blended-ideas** [comment]
You cannot cancel a promise,
However you can convert the promise being returned to an Observable, and treat it the same way you treat an Http Call.

**Comment by abhijeetgiram** [comment]
@karthikrp 
Hello,

I have solved my issue using the following way =>

const myObservable = Observable.fromPromise(
      this.odooRPC.call('model_name, 'method_name', parameter, {})
    );

const mySubscription = myObservable .subscribe(
   res => {
   console.log('res success');
   }, err => {
   console.log(err failed');
   }
);

setTimeout(() => {
mySubscription .unsubscribe();
}, 3000);

Need Help. Is it the right way?
Because "myObservable.timeout(3000)" is not working. I have tried to use it in the same HTTP fashion. But it does not work.
